server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -165,7 +166,9 @@ protected Executor getExecutor(GetRequest request, ShardId shardId) {
         final ClusterState clusterState = clusterService.state();
         if (clusterState.metadata().getIndexSafe(shardId.getIndex()).isSystem()) {
             return threadPool.executor(executorSelector.executorForGet(shardId.getIndexName()));
-        } else if (indicesService.indexServiceSafe(shardId.getIndex()).getIndexSettings().isSearchThrottled()) {
+        }
+        var indexService = indicesService.indexService(shardId.getIndex());
```

Comment:
We used to get `IndexNotFoundException` on relocations here which is not directly related to hollow shards, but something that was discovered during of stress testing of real-time gets.